crates/sui-indexer/src/models_v2/objects.rs

```diff
@@ -293,3 +293,101 @@ impl From<CoinBalance> for Balance {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use move_core_types::{account_address::AccountAddress, language_storage::StructTag};
+    use sui_types::{
+        coin::Coin,
+        digests::TransactionDigest,
+        gas_coin::{GasCoin, GAS},
+        object::{Data, MoveObject, Owner},
+        Identifier, TypeTag,
+    };
+
+    use super::*;
+
+    #[test]
+    fn test_canonical_string_of_object_type_for_coin() {
+        let test_obj = Object::new_gas_for_testing();
+        let indexed_obj = IndexedObject::from_object(1, test_obj, None);
+
+        let stored_obj = StoredObject::from(indexed_obj);
+
+        match stored_obj.object_type {
+            Some(t) => {
+                assert_eq!(t, "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0x0000000000000000000000000000000000000000000000000000000000000002::sui::SUI>");
+            }
+            None => {
+                panic!("object_type should not be none");
+            }
+        }
+    }
+
+    #[test]
+    fn test_convert_stored_obj_to_sui_coin() {
+        let test_obj = Object::new_gas_for_testing();
+        let indexed_obj = IndexedObject::from_object(1, test_obj, None);
+
+        let stored_obj = StoredObject::from(indexed_obj);
+
+        let sui_coin = SuiCoin::try_from(stored_obj).unwrap();
+        assert_eq!(
+            sui_coin.coin_type,
+            "0x0000000000000000000000000000000000000000000000000000000000000002::sui::SUI"
+        );
+    }
+
+    #[test]
+    fn test_one_more_thing() {
```

Comment:
Bit of a weird test name -- what are you actually testing here, can you name it for that?

Reply:
thanks renamed